clients/web/test/spec/widgetsSpec.js

```diff
@@ -5,15 +5,25 @@
  */
 girderTest.startApp();
 
-function _setProgress(test, duration) {
+function _setProgress(test, duration, resourceId, resourceName) {
     /* Set or update a current progress notification.
      *
      * :param test: test parameter to send to the webclienttest/progress
      *     endpoint
      * :param duration: duration to send to the endpoint
+     * :param resourceId: optional resource ID that the progress notification
+     *     is associated with.
+     * : param resourceName: optional resource type that the progress notification
+     *     is associated with.
      */
     girder.rest.restRequest({path: 'webclienttest/progress', type: 'GET',
-        data: {test: test, duration: duration}});
+        data: {
+            test: test,
+            duration: duration,
+            resourceId: resourceId,
```

Comment:
The API supports a partial or complete resource, as long as an id is provided (so we have somewhere to link). The reason for this is different resources might have different requirements, i.e. a job wants to pass along its id but maybe not the `logs` field which can have a lot of data.

For that reason, I opted to just provide the id when testing.